packages/astro/src/core/errors/README.md

```diff
@@ -23,25 +43,43 @@ As long as it is unique, the exact error code used is unimportant. For example,
 
-### CLI specifics:
+### CLI specifics tips:
 - If the error happened **during an action that changes the state of the project** (ex: editing configuration, creating files), the error should **reassure the user** about the state of their project (ex: "Failed to update configuration. Your project has been restored to its previous state.")
 - If an "error" happened because of a conscious user action (ex: pressing CTRL+C during a choice), it is okay to add more personality (ex: "Operation cancelled. See you later, astronaut!"). Do keep in mind the previous point however (ex: "Operation cancelled. No worries, your project folder has already been created")
 
 ### Shape
 - **Error codes and names are permanent**, and should never be changed, nor deleted. Users should always be able to find an error by searching, and this ensures a matching result. When an error is no longer relevant, it should be deprecated, not removed.
-- Contextual information may be used to enhance the message or the hint. However, the error code itself should not be included in the message as it will already be shown as part of the the error.
-- Do not prefix `message` and `hint` with descriptive words such as "Error:" or "Hint:" as it may lead to duplicated labels in the UI / CLI.
+- Contextual information may be used to enhance the message or the hint. However, the code that caused the error or the position of the error should not be included in the message as they will already be shown as part of the error.
+- Do not prefix `title`, `message` and `hint` with descriptive words such as "Error:" or "Hint:" as it may lead to duplicated labels in the UI / CLI.
+
+### Documentation support through JSDoc
+
+Using JSDoc comments, [a reference for every error message](https://docs.astro.build/en/reference/error-reference/) is built automatically on our docs. Users can then search for a error code to find more information on how to fix the error they encountered.
+
+Here's how to create and format the comments:
+
+```js
+/**
+  * @docs <- Needed for the comment to be used for docs
+  * @message <- (Optional) Clearer error message to show in cases where the original one is too complex (ex: because of conditional messages)
+  * @see <- List of additional references users can look at
+  * @description <- Description of the error
+  */
+```
+Example:
+```js
+/**
+  * @docs
+  * @message Route returned a `returnedValue`. Only a Response can be returned from Astro files.
+  * @see
+  * - [Response](https://docs.astro.build/en/guides/server-side-rendering/#response)
+  * @description
+  * Only instances of [Response](https://developer.mozilla.org/en-US/docs/Web/API/Response) can be returned inside Astro files.
+  */
+```
+
+For `@message`, the property is intended to provider a slightly cleaned error message or if there's multiple messages possible, a collection of common messages. If possible, avoid doing substantial changes to messages as some users might search for the error message
```

Comment:
```suggestion
The `@message` property is intended to provide slightly more context when it is helpful: a more descriptive error message or a collection of common messages if there are multiple possible error messages. Try to avoid making substantial changes to existing messages so that they are easy to find for users who copy and search the exact content of an error message.
```
Just some other thoughts here, for your consideration!